### DISCUSSION TITLE
Use Coverity to scan v.c

### DISCUSSION THREAD

**Discussion by CecilioGovain** [vlang/v#11858] [Ideas]
[Coverity](https://scan.coverity.com/o/oss_success_stories) offers free scan to open source projects, for example [btrfs-next](https://scan.coverity.com/projects/btrfs-next) 
Please register [vc](https://github.com/vlang/vc) to [Coverity Scan](https://scan.coverity.com/projects) .

**Comment by tandr** [comment]
With not-so-through settings (and possibilities for false positive) - first brief pass gives not a perfect result for the generated code. There are some leaks (for example, time__wait leaks what's get allocated by memdup), bunch of negative index possibilities (for example  v_fixed_index(child_index, 11) could return -1 on line 15384), whole lot of unreachable code etc

So, if getting these errors out of generated code is of any priority, getting Coverity Scan to work on this code is highly recommended.

**Comment by CecilioGovain** [comment]
[PVS-Studio](https://pvs-studio.com/en/order/open-source-license/) team likes open projects very much. That's why, if you are working on such projects, we are ready to give you a free license for one year.

[Cppcheck](http://cppcheck.sourceforge.net/) uses unsound flow sensitive analysis. Several other analyzers use path sensitive analysis based on abstract interpretation, that is also great however that has both advantages and disadvantages. In theory by definition, it is better with path sensitive analysis than flow sensitive analysis. But in practice, it means Cppcheck will detect bugs that the other tools do not detect.

**Comment by CecilioGovain** [comment]
[SonarCloud](https://sonarcloud.io/pricing) is for everyone
> Free for public projects
>     Unlimited lines of code
>     Anyone can see your project and browse your code
>     You have access to the full SonarCloud feature set
